server/routes/pinIPFS.ts

```diff
@@ -0,0 +1,31 @@
+import { Request, Response, NextFunction } from 'express';
+import { DB, sequelize } from '../database';
+import ipfs from '../util/ipfs';
+export const Errors = {
+  NotLoggedIn: 'Not logged in',
+  InvalidAddress: 'Invalid address',
+  NoBlobPresent: 'No JSON blob was input',
+};
+
+const pinIPFS = async (
+  models: DB,
+  req: Request,
+  res: Response,
+  next: NextFunction
+) => {
+  if (!req.user) return next(new Error(Errors.NotLoggedIn));
+  if (!req.body.address_id) return next(new Error(Errors.InvalidAddress));
+  if (!req.body.blob) return next(new Error(Errors.NoBlobPresent));
```

Comment:
Similarly, if the user is sending us a JSON blob, we should validate that it's actually proper JSON (of the correct length), and not some sort of attack, before we forward it along to Pinata.

Reply:
Done

Reply:
<img width="674" alt="image" src="https://user-images.githubusercontent.com/11271352/157537943-2f2e0825-28d3-4475-95d1-04a0caa75a02.png">